lib/spack/spack/installer.py

```diff
@@ -86,6 +86,16 @@
 STATUS_REMOVED = "removed"
 
 
+def _write_timer_json(pkg, timer, cache):
+    extra_attributes = {"name": pkg.name, "cache": cache}
+    try:
```

Comment:
this try really shouldn't be required, but for some reason the `_install_from_cache` routine doesn't actually install anything in the `test_install_from_cache_ok` test.